ee/tabby-webserver/src/oauth/mod.rs

```diff
@@ -0,0 +1,65 @@
+use std::sync::Arc;
+
+use axum::{
+    extract::{Query, State},
+    http::StatusCode,
+    routing, Json, Router,
+};
+use hyper::{client::HttpConnector, Client};
+use hyper_rustls::HttpsConnector;
+use serde::Deserialize;
+use tracing::error;
+
+use crate::schema::{
+    auth::{AuthenticationService, GithubAuthError, GithubAuthResponse},
+    ServiceLocator,
+};
+
+pub(crate) mod github;
+
+#[derive(Clone)]
```

Comment:
instead of marking this as clone, does it better to wrap this state inside of `Arc`?